genDocs

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env node
+
+/**
+ * This example script expects a path do components/plugins as input,
+ * generating MD documentation files to docs/developer-guide/reference
+ * e.g. node genDocs web/client/components
+ */
+
+var fs = require('fs');
+var path = require('path');
+var docsToMarkdown = require('react-docs-markdown');
+var reactDocs = require('react-docgen');
+var spawn = require('child_process').spawn;
+var toc = require('markdown-toc');
+var tocString = "";
+var dir = process.argv[2];
+filelist = [];
+docFileList = [];
+var fileContent = "";
+
+var walkSync = function(dir, filelist) {
+  files = fs.readdirSync(dir);
+  files.forEach(function(file) {
+      if (fs.statSync(path.join(dir, file)).isDirectory()) {
+          filelist = walkSync(path.join(dir, file), filelist);
+      }
+      else {
+          // FIXME: check only jsx files?
+          checkExt(path.join(dir, file), '.jsx')
+      }
+  });
+  // Create TOC at end
+  tocContent = "TOC\n---\n\n"
+  docFileList.forEach(function(file) {
+    var name = getComponentName(file);
+    tocString += toc('# ' + name + '\n\n').content + '\n';
+  });
+  tocContent += tocString;
+  // FIXME: static output folder?
+  fs.writeFile('docs/developer-guide/reference/index.md', tocContent, (err) => {
+      if (err) throw err;
+    });
+}
+
+function checkExt(file, ext) {
+    if (path.extname(file) == ext) {
+        //filelist.concat(file);
+        const docgen = spawn('react-docgen', [ file ]);
```

Comment:
Why do we spawn an external process, we already have the react parser in memory with require, we should use that (by the way, this doesn't work on Windows)?